src/packet/ipv4.rs

```diff
@@ -39,29 +41,67 @@ pub struct Ipv4 {
     payload: Vec<u8>,
 }
 
-/// Calculates the checksum of an IPv4 packet
-pub fn checksum(packet: &Ipv4Packet) -> u16be {
-    use packet::Packet;
+/// Represents an IPv4 Packet
+///
+/// 0          8          16          24         32
+/// +---------------------------------------------+
+/// |                  source_ip                  |
+/// +----------------------+----------------------+
+/// |                destination_ip               |
+/// +-----------+----------+----------------------+
+/// |   zeros   | protocol | tcp/udp length       |
+/// +-----------+---------------------------------+
+#[packet]
+pub struct Ipv4PseudoHeader {
+    #[construct_with(u8, u8, u8, u8)]
+    source: Ipv4Addr,
+    #[construct_with(u8, u8, u8, u8)]
+    destination: Ipv4Addr,
+    zeros: u8,
+    #[construct_with(u8)]
+    next_level_protocol: IpNextHeaderProtocol,
+    inner_packet_length: u16be,
+    // just to avoid the compiler to complain
```

Comment:
No need for this comment... Though perhaps a bug report?

The idea of having the payload field was to support chaining different packet types together, it definitely feels out of place for certain packets though. I guess it will be investigated properly alongside issue #140.